lms/djangoapps/course_api/blocks/tests/test_api.py

```diff
@@ -77,3 +77,22 @@ def test_sub_structure(self):
                 self.assertIn(unicode(block.location), blocks['blocks'])
             else:
                 self.assertNotIn(unicode(block.location), blocks['blocks'])
+
+    def test_filtering_by_block_types(self):
+        sequential_block = self.store.get_item(self.course.id.make_usage_key('sequential', 'sequential_y1'))
+
+        # not filtered blocks
+        blocks = get_blocks(self.request, sequential_block.location, self.user, requested_fields=['type'])
+        self.assertEquals(len(blocks['blocks']), 5)
+        found_not_problem = False
+        for key in blocks['blocks']:
```

Comment:
Nit: Python provides a nice utility (`iteritems`) for iterating through all key, value pairs in a dict, to avoid the performance hit of retrieving the item again in your loop.  So it would be:
```
for key, block in blocks['blocks'].iteritems()
    if block['type'] != 'problem'
```
And, since you don't even need the `key` in the loop, you can just use `itervalues`:
```
for block in blocks['blocks'].itervalues()
    if block['type'] != 'problem'
```